var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -0,0 +1,410 @@
+    variant('exo2mat', default=False,
+            description='build exo2mat Matlab translator')
+    variant('exo_format', default=False,
+            description='build exo_format exodus format query')
+    variant('exodiff', default=False,
+            description='build exodiff')
+    variant('exodus', default=False,
+            description='build exodus library (C and Fortran)')
+    variant('exomatlab', default=False,
+            description='build exomatlab Matlab global variable translator')
+    variant('exotec2', default=False,
+            description='build exotec2')
+    variant('exotxt', default=False,
+            description='build exotxt translator')
+    variant('explore', default=False,
+            description='build explore (previously grope) application')
+    variant('fastq', default=False,
+            description='build fastq 2D mesh generator')
+    variant('gen3d', default=False,
+            description='build gen3d')
+    variant('genshell', default=False,
+            description='build genshell')
+    variant('gjoin', default=False,
+            description='build gjoin (use ejoin if possible)')
+    variant('grepos', default=False,
+            description='build grepos')
+    variant('ioss', default=False,
+            description='build IOSS library')
+    variant('mapvar', default=False,
+            description='build mapvar')
+    variant('mapvar-kd', default=False,
+            description='build mapvar with kd search')
+    variant('mat2exo', default=False,
+            description='build mat2exo Matlab translator')
+    variant('nemesis', default=False,
+            description='build nemesis library (deprecated)')
+    variant('nemslice', default=False,
+            description='build nemslice')
+    variant('nemspread', default=False,
+            description='build nemspread')
+    variant('numbers', default=False,
+            description='build numbers')
+    variant('plt', default=False,
+            description='build plt library')
+    variant('svdi', default=False,
+            description='build svdi library')
+    variant('slice', default=False,
+            description='build slice (experimental parallel decomposition)')
+    variant('supes', default=False,
+            description='build supes library (internal use)')
+    variant('suplib', default=False,
+            description='build suplib library (internal use)')
+    variant('txtexo', default=False,
+            description='build txtexo translator')
+
+    # ###################### Dependencies ##########################
+
+    # Everything should be compiled position independent (-fpic)
+
+    depends_on('netcdf@4.6.2+mpi+parallel-netcdf', when='+mpi')
+    depends_on('netcdf@4.6.2~mpi', when='~mpi')
+    depends_on('cgns@develop+mpi+scoping', when='+cgns +mpi')
+    depends_on('cgns@develop~mpi+scoping', when='+cgns ~mpi')
+    depends_on('matio', when='+matio')
+    depends_on('zlib', when="+zlib")
```

Comment:
Unless I'm going blind, I don't see a zlib variant.

Reply:
Aren't all of these dependencies also needed when `+allpkgs`?

Reply:
The `matio` and `cgns` aren't packages but there are packages which depend on on the presence or absence of these variants.  For example, the `exo2mat` and `mat2exo` won't be built if `matio` is not present, but the cmake logic figures that out, so I don't know if I should also have that logic in place in the `package.py`.

For `cgns` and `metis`, the `IOSS` library will have additional capabilities if those libraries are present, but will still build without those libraries; just with reduced capability.

I removed the`zlib` variant since it is not a direct dependency of `seacas`; it is a dependency of `hdf5` which comes through either `netcdf` and/or `cgns` so shouldn't be specified at the `seacas` level.